### PR TITLE
Use non default ports

### PR DESCRIPTION
**Pull request by girisairam450**

### CODE REVIEW

File: swarm-learning/bin/common [hunk 1]
```diff
@@ -255,7 +255,21 @@ genDockerExec()
               "${3}" != "${containerFQDN}"     && \
               "${3}" != "${containerHostname}" ]]
         then
-            ssh="ssh ${3}"
+            # ssh="ssh ${3}"
+            if [[ -n "${5}" ]]
```
Comment (ravi-ss) [inline swarm-learning/bin/common:259]: Update the comments for this function

File: swarm-learning/bin/common [hunk 2]
```diff
@@ -393,7 +407,15 @@ processCommonOpt()
             checkAndAssign "${opt}" "${optarg}"
             workdir="--workdir=${optarg}"
             ;;
-
+        
+        --ssh-user)
```
Comment (ravi-ss) [inline swarm-learning/bin/common:411]: Update the help text.

Is no change required to the README files?

**Review by ravi-ss — CHANGES_REQUESTED** [review]

File: swarm-learning/bin/common (hunk 1 above)
Comment (suresh-ls) [reply swarm-learning/bin/common:259]: Yes, its unclear what are $5 and $6.

File: swarm-learning/bin/common (hunk 2 above)
Comment (suresh-ls) [reply swarm-learning/bin/common:411]: - How  is this  --ssh-user different from the -user above ??

- Do we need a short name for this, just for uniformity as all others have short forms..

**Review by suresh-ls — APPROVED** [review]

File: swarm-learning/bin/common (hunk 1 above)
Comment (girisairam450) [reply swarm-learning/bin/common:266]: added appropriate comments

File: swarm-learning/bin/common (hunk 2 above)
Comment (girisairam450) [reply swarm-learning/bin/common:419]: @suresh-ls 
--ssh-user is for connecting to spire server via this user. While --user is within the given node. run-sn uses long forms while init-workspace uses short forms. So uniformity is met.
@ravi-ss 
updated the read me and help text of this function.

File: docs/RunningSL.md [hunk 3]
```diff
@@ -261,6 +261,18 @@ Use the ``swarm-learning/bin/run-sn`` script to start Sentinel and Swarm Network
     
     Default: None
 
+-   ``--ssh-port <ssh port number>``
```
Comment (ravi-ss) [inline docs/RunningSL.md:264]: Fix alignment on lines 107-109 and 358-362.

File: docs/RunningSL.md [hunk 4]
```diff
@@ -261,6 +261,18 @@ Use the ``swarm-learning/bin/run-sn`` script to start Sentinel and Swarm Network
...
+    This parameter is optional. When it is not specified, connection to SPIRE Server happens via default port. This parameter is required when the SPIRE Server uses 
+    non-default ssh port. Swarm Network nodes uses this port to ssh connect SPIRE Server.
+    
+    Default: 22
+
+-   ``--ssh-user <ssh user name>``
+
+    This parameter is also optional. When it is specified, connection to SPIRE Server happens via specified user.
+    Swarm Network nodes uses this user to ssh connect SPIRE Server. 
+
```
Comment (ravi-ss) [inline docs/RunningSL.md:275]: This should be rephrased and documented under the common options

File: swarm-learning/bin/common [hunk 5]
```diff
@@ -213,7 +213,11 @@ isLocalhost()
 #      the remote host. Further, the user should have configured a password-less
 #      SSH login to the remote system (including a login for the root user, when
 #      sudo is required to run docker) for this to work.
-#   4. A descriptive name for the image - to include in error messages.
+#   4. Optional ssh port and ssh user. If not specified the connection will be
```
Comment (ravi-ss) [inline swarm-learning/bin/common:216]: This is not correct. This should be:
   * 4 - descriptive name
   * 5 - ssh port
   * 6 - ssh user

File: swarm-learning/bin/common (hunk 2 above)
Comment (ravi-ss) [reply swarm-learning/bin/common:419]: You should document these new options in the usage function

**Review by ravi-ss — CHANGES_REQUESTED** [review]

File: swarm-learning/bin/common [hunk 6]
```diff
@@ -214,6 +214,11 @@ isLocalhost()
 #      SSH login to the remote system (including a login for the root user, when
 #      sudo is required to run docker) for this to work.
 #   4. A descriptive name for the image - to include in error messages.
+#   5. Optional ssh port. If not specified the connection will be 
+#      established using the default port 22.
+#   6. Optional ssh user. On specifying the user name, the ssh connection will 
+#      happen as the mentioned user.
```
Comment (girisairam450) [inline swarm-learning/bin/common:220]: add if not specified it will be taken as current user.

File: swarm-learning/bin/common (hunk 2 above)
Comment (girisairam450) [reply swarm-learning/bin/common:420]: add it to printCommonUsage

File: docs/RunningSL.md (hunk 4 above)
Comment (girisairam450) [reply docs/RunningSL.md:275]: This parameter specifies the port number on which the SSH server is listening. It is optional. 
It should be specified in a multi-system cluster environment when the SSH servers are not listening on the default SSH port.

This parameter specifies the user name to use when establishing an SSH connection. It is optional. When it is not specified, the current user's login name is used.

File: docs/Prerequisites.md [hunk 7]
```diff
@@ -78,5 +78,8 @@ cluster. The requirements for each system are:
 -   Multi system cluster Requirements:
 
     -   Password-less SSH setup across systems
-
+    
+        >   *NOTE*: If needed, a non-default port SSH connection is possible through ssh port and ssh user options.
```
Comment (girisairam450) [inline docs/Prerequisites.md:82]: Not needed

File: swarm-learning/bin/common (hunk 2 above)
Comment (girisairam450) [reply swarm-learning/bin/common:428]: added to common usage

File: docs/RunningSL.md (hunk 4 above)
Comment (girisairam450) [reply docs/RunningSL.md]: rephrased.

**Review by suresh-ls — APPROVED** [review]
Looks good.

File: swarm-learning/bin/common [hunk 8]
```diff
@@ -116,7 +116,15 @@ printCommonUsage()
     printf -- "\tWorking directory inside the container\n\n"
     
     printf -- "--rm\n"
-    printf -- "\tRequest Docker to automatically remove the container when it exits\n\n"    
+    printf -- "\tRequest Docker to automatically remove the container when it exits\n\n"
+    
+    printf -- "--ssh-port\n"
+    printf -- "\tPort on which connection establishes to ssh listener\n"
+    printf -- "\tDefault: 22, if ssh listener doesn't use any custom port\n\n"
```
Comment (ravi-ss) [inline swarm-learning/bin/common:123]: The port on which the SSH server is listening.
Default: 22

File: swarm-learning/bin/common [hunk 9]
```diff
@@ -116,7 +116,15 @@ printCommonUsage()
...
+    
+    printf -- "--ssh-user\n"
+    printf -- "\tConnects to ssh listener using specified user\n"
+    printf -- "\tDefault: current user, if not specified\n\n"
```
Comment (ravi-ss) [inline swarm-learning/bin/common:127]: The user name to use when establishing a SSH connection.
Default: current user

**Review by ravi-ss — APPROVED** [review]

File: swarm-learning/bin/common (hunk 8 above)
Comment (girisairam450) [reply swarm-learning/bin/common]: updated

File: swarm-learning/bin/common (hunk 9 above)
Comment (girisairam450) [reply swarm-learning/bin/common]: updated

**Pull request merged**